Main.cc

```diff
@@ -567,19 +579,30 @@ int main(int argc, const char* argv[]) {
 
         session_number = 0;
 
-        websocket_hub.onMessage(&OnMessage);
-        websocket_hub.onConnection(&OnConnect);
-        websocket_hub.onDisconnection(&OnDisconnect);
-        websocket_hub.onError(&OnError);
-
-        if (websocket_hub.listen(port)) {
-            fmt::print("Listening on port {} with root folder {}, base folder {}, {} threads in worker thread pool and {} OMP threads\n",
-                port, root_folder, base_folder, thread_count, omp_thread_count);
-            websocket_hub.run();
-        } else {
-            fmt::print("Error listening on port {}\n", port);
-            return 1;
-        }
+        uWS::App()
+            .ws<PerSocketData>("/*", {.compression = uWS::SHARED_COMPRESSOR,
+                                         .maxPayloadLength = 16 * 1024,      // use default setting
```

Comment:
if we use designated initialisers, then we can leave out those properties that are initialised to default, can't we?

note that designated initialisers are part of the c++20 spec, but are supported on all our target compilers, as far as I can see

Reply:
It seems to me still got error messages on CentOS (using gcc8.4) if I remove `.maxPayloadLength`, `.idleTimeout` and `.maxBackpressure` settings.

Reply:
does it work on clang? @ajm-asiaa what version of GCC do we build with?

Reply:
Yes, it works on clang.

Reply:
We usually use GCC 8.4.0